File: corecomply-mvp/api/CoreComply.Api/Domain/Entities/Asset.cs
namespace CoreComply.Api.Domain.Entities;

public class Asset
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Type { get; set; } = string.Empty; // hardware, software, data, facility
    public string Classification { get; set; } = "internal"; // public, internal, confidential, restricted
    public string Owner { get; set; } = string.Empty;
    public string Location { get; set; } = string.Empty;
    public string Status { get; set; } = "active"; // active, inactive, retired
    public DateTime? PurchaseDate { get; set; }
    public decimal? Value { get; set; }
    public string? Description { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

File: corecomply-mvp/web/src/types/asset.ts
export interface Asset {
  id: number;
  name: string;
  type: string;
  classification: string;
  owner: string;
  location: string;
  status: string;
  purchaseDate?: string | null;
  value?: number | null;
  description?: string | null;
  createdAt: string;
}

export interface CreateAssetRequest {
  name: string;
  type: string;
  classification?: string;
  owner: string;
  location: string;
  status?: string;
  purchaseDate?: string | null;
  value?: number | null;
  description?: string | null;
}

export interface UpdateAssetRequest {
  name: string;
  type: string;
  classification: string;
  owner: string;
  location: string;
  status: string;
  purchaseDate?: string | null;
  value?: number | null;
  description?: string | null;
}
